code/modules/hydroponics/grown/melon.dm

```diff
@@ -39,6 +40,38 @@
 /obj/item/food/grown/watermelon/make_dryable()
 	return //No drying
 
+/obj/item/food/grown/watermelon/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/kitchen/spoon))
+		return ..()
+
+	var/melon_pulp_count = 1
+	if(seed)
+		melon_pulp_count += round(seed.potency / 25)
+
+	user.balloon_alert(user, "scooped out [melon_pulp_count] pulp(s)")
+	for(var/i in 1 to melon_pulp_count)
+		new /obj/item/food/watermelonmush(user.loc)
+
+	/// The piece of armour melon turns into; either chetsplate or helmet
+	var/melon_armour
```

Comment:
for clarity's sake this is what I meant by typing. Even if it has no value this makes a promise to people reading the code about what it will be
```suggestion
	var/obj/item/clothing/melon_armour
```

Reply:
what promise?